acq4/devices/MicroManagerCamera/mmcamera.py

```diff
@@ -152,21 +157,26 @@ def _readAllParams(self):
         with self.camLock:
             params = OrderedDict([(n, None) for n in defaultParams])
 
-            properties = self.mmc.getDevicePropertyNames(self.camName)
+            properties = self.mmc.getDevicePropertyNames(self.camName) + ('Exposure',) # because the CellCam driver didn't present the exposure as a property, need to add it with a getExposure() call
             for prop in properties:
                 vals = self.mmc.getAllowedPropertyValues(self.camName, prop)
                 if vals == ():
-                    if self.mmc.hasPropertyLimits(self.camName, prop):
-                        vals = (
-                            self.mmc.getPropertyLowerLimit(self.camName, prop),
-                            self.mmc.getPropertyUpperLimit(self.camName, prop),
-                        )
+                    if prop != 'Exposure': # workaround for exposure working
+                        if self.mmc.hasPropertyLimits(self.camName, prop):
+                            vals = (
+                                self.mmc.getPropertyLowerLimit(self.camName, prop),
+                                self.mmc.getPropertyUpperLimit(self.camName, prop),
+                            )
+                        else:
+                            # just guess..
+                            vals = (1e-6, 1e3)
                     else:
-                        # just guess..
-                        vals = (1e-6, 1e3)
+                        vals = (1, 100) # sensible range of exposure values...
                 else:
                     vals = list(vals)
-                readonly = self.mmc.isPropertyReadOnly(self.camName, prop)
+                if prop != 'Exposure': # again, workaround...
+                    readonly = self.mmc.isPropertyReadOnly(self.camName, prop)
```

Comment:
This is still valid code for non-CellCam cameras.

Reply:
Oh, also, `readonly` is used in code further along, so we would need a default value (I'm guessing `False`) for CellCam.

Reply:
Yeah, so we could perhaps add an else block.
```suggestion
                    readonly = self.mmc.isPropertyReadOnly(self.camName, prop)
                else:
                    readonly = False
```